Xamarin.PropertyEditing/IOrigin.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Xamarin.PropertyEditing.Drawing;
+
+namespace Xamarin.PropertyEditing
+{
+	public interface IOrigin
+	{
+		/// <summary>
+		/// Gets whether this property supports Origins or not
+		/// </summary>
+		/// <remarks>
+		/// If this is <c>true</c>, <see cref="CommonOrigin Origin"/> should contain a value.
+		/// </remarks>
+		bool HasOrigin { get; }
+
+		/// <summary>
+		/// Gets whether the value should be constrained to those that are predefined.
+		/// </summary>
+		/// <remarks>
+		/// Implemented on <see cref="IPropertyInfo"/> instances.
+		/// </remarks>
+		CommonOrigin Origin { get; }
```

Comment:
This would only allow for an origin to be predefined on a per-property basis and never changed. The origin needs  to be part of the value, whether by adding a nullable origin to common rectangle or adding a new type that joins the two.